respect-lib-shared/src/commonMain/kotlin/world/respect/shared/viewmodel/manageuser/profile/ProfileViewModel.kt

```diff
@@ -0,0 +1,165 @@
+                    userAccountIconVisible = false
+                )
+            }
+        }
+
+    }
+
+
+    fun onFullNameChanged(value: String) {
+        viewModelScope.launch {
+            _uiState.update {
+                it.copy(
+                    fullName = value,
+                    fullNameError = if (value.isNotBlank()) null else getString(Res.string.full_name_required)
+                )
+            }
+        }
+
+    }
+
+    fun onGenderChanged(value: OneRosterGenderEnum) {
+        viewModelScope.launch {
+            _uiState.update { prev ->
+                prev.copy(
+                    gender = value,
+                    genderError = if (value != OneRosterGenderEnum.UNSPECIFIED) null else getString(
+                        Res.string.gender_required
+                    )
+                )
+            }
+        }
+    }
+
+    fun onDateOfBirthChanged(value: LocalDate?) {
+
+        viewModelScope.launch {
+            _uiState.update { prev ->
+                prev.copy(
+                    dateOfBirth = value,
+                    dateOfBirthError = if (value != null) null else getString(Res.string.dob_required)
+                )
+            }
+        }
+    }
+
+    fun onClickSave() {
+
+        viewModelScope.launch {
+            val current = _uiState.value
+            _uiState.update { prev ->
+                prev.copy(
+                    fullNameError = if (current.fullName.isBlank()) getString(Res.string.full_name_required) else null,
+                    genderError = if (current.gender == OneRosterGenderEnum.UNSPECIFIED) getString(
+                        Res.string.gender_required
+                    ) else null,
+                    dateOfBirthError = if (current.dateOfBirth == null) getString(Res.string.dob_required) else null
+                )
+            }
+
+            val hasError = listOf(
+                current.fullName.isBlank(),
+                current.gender == OneRosterGenderEnum.UNSPECIFIED,
+                current.dateOfBirth == null
+            ).any { it }
+
+            if (hasError){
+                return@launch
+            }else{
+                when (route.type) {
+                    ProfileType.PARENT ,ProfileType.Student->{
+                        viewModelScope.launch {
+                            _navCommandFlow.tryEmit(
+                                NavCommand.Navigate(SignupScreen.create(route.type))
+                            )
+                        }
+                    }
+                    ProfileType.CHILD ->{
+                        viewModelScope.launch {
+                            _navCommandFlow.tryEmit(
+                                NavCommand.Navigate(WaitingForApproval)
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+data class ProfileUiState(
+    val screenTitle: String = "",
+    val actionBarButtonName: String = "",
```

Comment:
Labels and the like should be StringResource type if they are always string resources, otherwise use UiText